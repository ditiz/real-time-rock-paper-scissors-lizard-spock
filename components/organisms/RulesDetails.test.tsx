import "@testing-library/jest-dom"
import { render, screen } from "@testing-library/react"
import RulesDetailsOrganism from "./RulesDetails"

it("renders RulesDetailsOrganism unchanged", () => {
  const { container } = render(<RulesDetailsOrganism />)
  expect(container).toMatchSnapshot()
})

it("should contain image of rules", () => {
  render(<RulesDetailsOrganism />)
  expect(screen.getByRole("img")).toBeInTheDocument()
})

it("should contain rules", () => {
  render(<RulesDetailsOrganism />)
  expect(screen.getByText("Scissors cuts Paper")).toBeInTheDocument()
  expect(screen.getByText("Spock vaporizes Rock")).toBeInTheDocument()
})
