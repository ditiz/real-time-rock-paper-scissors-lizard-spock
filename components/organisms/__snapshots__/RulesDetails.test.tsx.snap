// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders RulesDetailsOrganism unchanged 1`] = `
<div>
  <section
    class="rules__details"
  >
    <div>
      <img
        alt="rules"
        data-nimg="1"
        decoding="async"
        height="500"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=1080&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 1x, /_next/image?url=%2Fimg.jpg&w=1080&q=75 2x"
        style="color: transparent;"
        width="500"
      />
    </div>
    <div>
      <p
        class="text"
      >
        Scissors cuts Paper
      </p>
      <p
        class="text"
      >
        Paper covers Rock
      </p>
      <p
        class="text"
      >
        Rock crushes Lizard
      </p>
      <p
        class="text"
      >
        Lizard poisons Spock
      </p>
      <p
        class="text"
      >
        Spock smashes Scissors
      </p>
      <p
        class="text"
      >
        Scissors decapitates Lizard
      </p>
      <p
        class="text"
      >
        Lizard eats Paper
      </p>
      <p
        class="text"
      >
        Paper disproves Spock
      </p>
      <p
        class="text"
      >
        Spock vaporizes Rock
      </p>
      <p
        class="text"
      >
        (and as it always has) Rock crushes Scissors
      </p>
    </div>
  </section>
</div>
`;
